frontend/src/pages/Students/StudentProfilePage.js

```diff
@@ -3,27 +3,64 @@ import {
   getStudentDetail,
   followStudent,
   unfollowStudent,
+  getStudentActivityLog,
 } from "../../redux/actions/student";
 import { useDispatch, useSelector } from "react-redux";
-import { useParams } from "react-router";
+import { useParams, useHistory } from "react-router";
 import Loading from "../../components/common/Loading";
 import ActivityFeed from "../../components/common/ActivityFeed";
 import { tempStrFunction } from "../../utils";
 
 const StudentProfilePage = () => {
   const dispatch = useDispatch();
-  const { student, total_followers, total_following, is_following } =
-    useSelector((state) => state.students);
+  const {
+    student,
+    total_followers,
+    total_following,
+    is_following,
+    activityLogs,
+  } = useSelector((state) => state.students);
   const { id } = useParams();
+  const history = useHistory();
+  const { userData } = useSelector((state) => state.auth);
 
   useEffect(() => {
     dispatch(getStudentDetail(id));
-  }, [dispatch, id]);
+    dispatch(getStudentActivityLog(id));
+    userData.user_id === parseInt(id) && history.push("/");
+  }, [dispatch, id, history, userData]);
 
   const followStudentAction = () => {
     is_following ? dispatch(unfollowStudent(id)) : dispatch(followStudent(id));
   };
 
+  const filteredActivityLog = () => {
+    let ownactivityLog = [];
+    activityLogs &&
+      activityLogs.map((log) => {
+        if (log.resourcetype === "StudentFollowInformation") {
+          log.follower.id === parseInt(id) && ownactivityLog.push(log);
+        } else {
+          log.student.id === parseInt(id) && ownactivityLog.push(log);
+        }
+        return null;
+      });
```

Comment:
Since we can use map, no need to push unto array. Suggestion:
```suggestion
    const ownactivityLog = activityLogs.map((log) => {
      if (log.resourcetype === "StudentFollowInformation") {
        if (log.follower.id === parseInt(id)) return log;
      } else {
        if (log.student.id === parseInt(id)) return log;
      }
    });
```

But I also observed you are returning null in line 46. So perhaps what you need is the `filter` function, instead of map?